### ISSUE TITLE
Wrong Answer focusing on image meta properties from MLLM

### ISSUE THREAD

**Issue by xp1632** [xp1632/Aalen_working_log#26]
- We find this issue from here when testing the ability of blip2 to infer correct image meta properties 
- https://github.com/xp1632/Aalen_working_log/issues/25#issuecomment-2710059984
---

**3.	Wrong answer from Vision Language Model about image property (grayscale or RGB)**
-	We find out that Blip2 answers wrongly about image meta properties
-	After testing with `rephrasing questions`, `if lose information in preprocessing image`
-	The most possible reason is `model was trained focusing on semantic features`
-	We can use OpenCV to compensate the correct meta properties

---

- [ ]  **Question 0: Did Blip2 answer other properties wrongly?**
     - test Blip2 with validation question set that covers more meta properties
- [ ] Question 1: Did other MLLM model have similar issue?
     - we'll choose from: Llava, gemmi3, Llama, qwen 
- [ ] Question 2: If this gap existed,  what did other researchers do for this?
     - we find Q-instruct, which contribute low-level perception dataset and fine-tuning for mllm

**Comment by xp1632** [comment]
**Feedback 1:**

1.	- [ ] Check publications about how vision language model lacking the ability of perceiving correct image basic properties
(Gaps and Solutions) 

2.	- [ ] Create a validation Q&A set and check if other vision language models behave like blip2 model (giving wrong answer about RGB etc.)

3.	- [ ] Synchronize with Fei to avoid repetitive work while his master student is doing similar work 

---

**Feedback 2:**

1. **Research Question:**

- Q1: Can MLLM models (**note 1**) find meta data (**note 2**) of image? 

  **note 1: MLLM models**: chosen from the list of MLLM that other papers also choose to compare with, such as Blip2, Llava, Qwen...
 **note 2: meta data**: even lower level of visual aspects, not the low-level but still context information in Q-Instruct paper such as (blurred, light condition...), but the more **technical meta details** of the image itself such as (resolution, bit depth, color channel, compression, aspect ratio... )

- Q2: Can this finding be improved by providing a callback query (**note 3**) for meta data?
  **note 3: callback query**: we provide a function where the model can use opencv libraries to acquire visual meta information of an image 

2. **Research Strategy**

- Contribution Niche Check:  
     - Please check carefully in the beginning,  that [Our Method] fits in the niche of  [State of Art], while either other people didn't do this particular type or theirs is worse than ours 

- Backwards Research:
     - We research with an inverse order:
     - [Result]   ----> [Experiment]    ----> [Method]
     - Think beforehand what kind of result you are expecting -->
     - What kind of experiment are you going to conduct --->
     - What method will you need
     - Examining the paper in the early stage as the role of a reviewer

- Use 'Traditional' Metric
    - If other researchers in this area choose certain metrics or models, go with them, save the creativity in other sections

3. Conference List:

 - EU/June, July, August/ Workshop or lower CVPR ones

**Comment by xp1632** [comment]
### Related Publications

- The first round of paper is not quite what we need; we need better keywords for searching
- We could try these keywords:

> "Technical image understanding in MLLMs"
> "MLLM for image quality assessment"
>  --->"**Low-level vision properties in large vision-language models"**

 ---


1.   (2022) **"Do Vision-Language Pretrained Models Learn Primitive Concepts?" - analyzes fundamental visual feature perception**
   - https://cs.brown.edu/research/pubs/theses/masters/2022/yun.bhalla.pdf
   - primitive concepts: such as color and shape attributes,
   - not so relevant
   - still semanticly (when learning this is {ripe}”, “this is {apple}”,  if  “this is {ripe apple} is learnt)
  - but we can learn its fashion of doing experiment on vl models 
---


2.  (2025) **“Color in Visual-Language Models: CLIP deficiencies”**

  - https://arxiv.org/html/2502.04470v1
   - not so relevant 
   - This paper checks how the color is perceived in clip model, specifically which neurons in which layer
   - and find vision language model focuses more on information from text instead of image by doing a Stroop experiment (text "red" in green color is recognized more likely as red instead of green)


---


3. (2025) "Grounding-IQA: Multimodal Language Grounding Model for Image Quality Assessment"
 - https://www.themoonlight.io/en/review/grounding-iqa-multimodal-language-grounding-model-for-image-quality-assessment

 - this paper is about using natural language description for image quality assessment instead of score

![Image](https://github.com/user-attachments/assets/1acfb9d0-288c-4604-a409-5bd5d44dd685)

----


### **4. **(2024_cvpr)** "Q-Instruct: Improving Low-level Visual Abilities for Multi-modality Foundation Models"**
- https://github.com/Q-Future/Q-Instruct/blob/main/fig/Q_Instruct_v0_1_preview.pdf
- Contribution: first of a kind multi-modal datasets on low-level visual aspects for low-level visual instruction tunning for mllms

![Image](https://github.com/user-attachments/assets/d4753476-07e4-4b66-8c01-b77a51641237)

- Q Pathway: human feedback dataset focusing on low-level visual aspects 
![Image](https://github.com/user-attachments/assets/f4ad61cf-71d2-46e5-9418-b83426bcb135)

- Q-Instruct:  Q&A pairs focusing on low-level visual aspects

![Image](https://github.com/user-attachments/assets/160fcae0-2cbf-4a33-8e8d-48f4de227ad5)

- Improvement of models on ability of low-level perception

![Image](https://github.com/user-attachments/assets/4d2d312a-5e4d-4713-80a0-cffd506f2824)


- Improvement of models on ability of low-level description 

![Image](https://github.com/user-attachments/assets/4cd49f89-9989-4c29-89af-4c6ef9654547)

---



### I found a great paper about mllm benchmarks:

**- MMMU: A Massive Multi-discipline Multimodal Understanding and Reasoning Benchmark for Expert AGI**
- paper: https://openaccess.thecvf.com/content/CVPR2024/papers/Yue_MMMU_A_Massive_Multi-discipline_Multimodal_Understanding_and_Reasoning_Benchmark_for_CVPR_2024_paper.pdf
- github: https://mmmu-benchmark.github.io

- and here are some state of art mllm:

![Image](https://github.com/user-attachments/assets/c1daa9d5-4b79-4e61-bed5-74662c6077ac)

**Comment by xp1632** [comment]
### Validation Question Set 
- for testing MLLM's abilities to infer image meta properties


- [x] 1. We start by image dataset [TID2013](https://huggingface.co/datasets/xmba15/TID2013/viewer/default/train?p=28&views%5B%5D=train),  [LIVE](https://paperswithcode.com/dataset/live1)

- [x]  2. we select the existing subset of image which contains metadata variation we need such as bit depth or just use some pictures as masters base

- [x]  3. we write python module that could generate images in specific varied condition


---

**- image testbed reference from Tim**

- basic image restaurations        
(1) linear brighness / contrast / hue / saturation adaptation
(2) non-linear tone curve adaptation
(3) geometric operations: translation, rotation, scaling, shearing
(4) perspective correction
(5) morphological operations: erosion, delation, white hat, black hat
(6) convolutions: blurring, edge detection

- combined image restaurations        
 (1) concattenate two basic operations
 (2) concattenate three basic operations
 (3) concattenate four basic operations
        
- maybe later: advanced inverse problems
 (1) super resolution
 (2) denoising
 (3) re-coloring


---

### Image Meta Property Items


**1. Basic Color Properties**

  - RGB vs. grayscale detection
  - Color depth/intensity
  - Presence of specific color channels


**2. Image Quality Attributes**

- Resolution (high vs. low)
- Blur detection
- Compression artifacts
- Noise levels


**3. Image Composition**

- Aspect ratio
- Framing (close-up vs. wide)
- Lighting conditions (dark, bright, backlit)


**4. Visual Structure**

- Focus areas (shallow vs. deep depth of field)
- Perspective properties
- Symmetry/asymmetry

**5. Technical Indicators**

- Image type (photograph, illustration, rendering)
- Post-processing indicators (filtered, HDR)
- Synthetic vs. natural image detection

----

**Comment by xp1632** [comment]
### Metrics of evaluation


- How should we evaluate the answer from MLLMs?
- we can refer to these two papers on model selection and criteria selection
- actually what we are doing is also building a benchmark, but 
- do remember to check the last few text of this https://claude.ai/chat/c6c2747e-62b1-4033-bfcd-4840f8011567 before start designing experiment
----

1. (cvpr 2024)SEED-Bench: Benchmarking Multimodal Large Language Models
 https://openaccess.thecvf.com/content/CVPR2024/papers/Li_SEED-Bench_Benchmarking_Multimodal_Large_Language_Models_CVPR_2024_paper.pdf
- moonlight version: https://www.themoonlight.io/paper/share/ba7be7e8-c162-4e95-ae1c-bac436bfadb2

![Image](https://github.com/user-attachments/assets/d3cd4c0f-f9e4-483c-86cf-9da1c2d841ac)



- Model choice:

![Image](https://github.com/user-attachments/assets/1f40275d-ac63-4eff-8668-164d8fbc6f22)

- multi-choice question generation pipeline:

![Image](https://github.com/user-attachments/assets/7dca9ead-bec3-4287-b43b-12accca67ec7)

- Evaluation strategy of text output: {answer ranking strategy instead of instruct ABCD} evaluates MLLMs by determining how likely the model is to produce each potential answer, independent of explicit instructions or the order of the answer choices. 

![Image](https://github.com/user-attachments/assets/0fbf7813-f747-4905-96e1-2b66c04419de)

![Image](https://github.com/user-attachments/assets/c2220171-cb2b-48e8-9742-27ed4c015ff3)

- Evaluation strategy of image output: compute CLIP feature similarity 

- Evaluation result image that i like:

![Image](https://github.com/user-attachments/assets/c118feeb-5683-4419-b796-91085335bd64)


----

2. (！)MME: A Comprehensive Evaluation Benchmark for Multimodal Large Language Models
          - https://arxiv.org/pdf/2306.13394     
                  
- image of comparison of 6 models on 14 tasks:

![Image](https://github.com/user-attachments/assets/7c2e769a-2167-493e-b5a9-f5af48a5e46f)

**Comment by xp1632** [comment]
### Solution framework:

[link](https://claude.ai/chat/edc8abc6-65ce-45c2-85df-faf04ad91b86)
![Image](https://github.com/user-attachments/assets/dd105716-035b-4d37-a933-4ca1dd6828d2)

![Image](https://github.com/user-attachments/assets/e6e06ff7-d0dd-4e80-a052-e31bb1b9f0e8)
